.changeset/dull-mails-fold.md

```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+transform sidebar headings into content span elements
```

Comment:
Small suggestion for the text hear to help make it clearer exactly what changed:

```suggestion
Use `<span>` instead of `<h2>` in sidebar group headings
```